### PR TITLE
Feature/add component tests

### LINKED ISSUE #47: Create Cypress component tests for existing UI components
## Overview of the Feature Request

Now that we have decided to use Cypress for our component testing,cwe need to add Cypress tests for the components that have already been created.

## What kind of user is the feature intended for?

This feature is intended for developers who use React to build and test components for our web applications.

## What inspired the request?

The analysis of the storybook testing capabilities #27 

## What existing behavior do you want changed?

We need to add Cypress tests for the components that have already been created.

## Any brand new behavior do you want to add to Dataverse?

No, this feature request does not introduce any brand new behavior to Dataverse.

## Any open or closed issues related to this feature request?

- https://github.com/IQSS/dataverse-frontend/issues/27
- https://github.com/IQSS/dataverse-frontend/issues/32

### PR DESCRIPTION
**Pull request by ekraffmiller**
## What this PR does / why we need it:
Creates Cypress component tests for existing UI components

## Which issue(s) this PR closes:

- Closes #47 

## Special notes for your reviewer:
The all the added components use i18n keys, except for localizedLayout.spec.tsx and helloDataverse.spec.tsx, which use English translation.  I thought it would be good to keep it that way to show how we can test either with keys or with translated text.

## Suggestions on how to test this:
npm run cy:open, then choose component testing to run the tests interactively, or npm run cy:run-component to run all the component tests from the command line.  (The component tests are also run in the github test action.)

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
No
## Is there a release notes update needed for this change?:
No

### CODE REVIEW

**Comment by MellyGray** [comment]
Looks good 😄 

But I'd like to continue using cypress-testing-library, so instead of relying on the data-testid or classes to find the elements rely on things the user can actually see. Like the text or role of the elements.

The testing library philosophy is that we create tests from the user point of view so the tests are unlikely to change with time. We can always change classes names or even the tag of an elemen or the data-tested, but we'll unlikely change the functionality with is what the user sees. Also testing library tests are more semantic. 

For example this test its hardly human readable:

```
it('renders dropdown links correctly', () => {
     cy.mount(<Navbar brand={brand} links={linksWithDropdown} />)
     cy.pause()
     cy.get('nav').should('exist')
     cy.get('nav .navbar-brand').should('have.attr', 'href', '/')
     cy.get('nav .navbar-brand img').should('have.attr', 'src', 'path/to/logo')
     cy.get('nav .navbar-brand img').should('have.attr', 'alt', 'logo')
     cy.get('nav .navbar-brand').should('contain', 'My Site')
     cy.get('nav .nav-link').should('have.length', 3)
     cy.get('nav .nav-link:eq(0)').should('have.attr', 'href', '/link1')
     cy.get('nav .nav-link:eq(0)').should('contain', 'Link 1')
     cy.get('nav .nav-link:eq(1)').should('have.attr', 'href', '/link2')
     cy.get('nav .nav-link:eq(1)').should('contain', 'Link 2')
     cy.get('nav .nav-item.dropdown').should('exist')
     cy.get('nav .nav-item.dropdown .nav-link').should('have.attr', 'role', 'button')
     cy.get('nav .nav-item.dropdown .nav-link').should('contain', 'Dropdown')
     cy.get('.navbar-toggler-icon').click()
     cy.get('nav .nav-item.dropdown').should('be.visible')
     cy.get('nav .nav-item.dropdown').click()
     cy.get('nav .nav-item.dropdown .dropdown-menu').should('exist')
     cy.get('nav .nav-item.dropdown .dropdown-menu .dropdown-item').should('have.length', 2)
     cy.get('nav .nav-item.dropdown .dropdown-menu .dropdown-item:eq(0)').should(
       'have.attr',
       'href',
       '/sublink1'
     )
     cy.get('nav .nav-item.dropdown .dropdown-menu .dropdown-item:eq(0)').should(
       'contain',
       'Sublink 1'
     )
     cy.get('nav .nav-item.dropdown .dropdown-menu .dropdown-item:eq(1)').should(
       'have.attr',
       'href',
       '/sublink2'
     )
     cy.get('nav .nav-item.dropdown .dropdown-menu .dropdown-item:eq(1)').should(
       'contain',
       'Sublink 2'
     )
   })

```

But with testing library we look for the role and text, which are unlikely to change, and they are more human readable than navbar classes, which could change and are hard to read.

```
test('renders the dropdown links', async () => {
    const { getByRole, findByRole } = render(<NavDropdown title="Dropdown Title" links={links} />)

    const dropdownTitle = getByRole('button', { name: 'Dropdown Title' })

    fireEvent.click(dropdownTitle)

    const link1Element = await findByRole('link', { name: 'Link 1' })
    expect(link1Element).toBeInTheDocument()

    const link2Element = await findByRole('link', { name: 'Link 2' })
    expect(link2Element).toBeInTheDocument()

    const link3Element = await findByRole('button', { name: 'Link 3' })
    expect(link3Element).toBeInTheDocument()

    fireEvent.click(link3Element)

    const sublink1Element = await findByRole('link', { name: 'Sublink 1' })
    expect(sublink1Element).toBeInTheDocument()

    const sublink2Element = await findByRole('link', { name: 'Sublink 2' })
    expect(sublink2Element).toBeInTheDocument()
  })
```

The example above is with vitest but is the same with cypress testing library.

What do you think?

File: .eslintrc.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "parserOptions": {
     "ecmaVersion": "latest",
     "sourceType": "module",
-    "project": ["./tsconfig.json", "./tests/e2e/tsconfig.json"]
+    "project": ["./tsconfig.json", "./cypress/tsconfig.json"]
```
Comment (MellyGray) [inline .eslintrc.json:32]: Why did you move the tests to a different folder? I like the `tests/`folders for the tests. Were there problems with the tsconfig or is it because a personal preference?

Comment (ekraffmiller) [reply .eslintrc.json:32]: Hi @MellyGray I updated the code to use the testing library, let me know what you think.  FYI, I tried using this assertion: 
` expect(cy.findByRole('link', { name: 'Sign Up' })).toBeInTheDocument()`
but it causes an error, related to an open issue in Cypress - https://github.com/cypress-io/cypress/issues/25894 , so that's why I'm using `should('exist')` instead.

File: src/sections/layout/footer/Footer.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ export function Footer() {
   const { t } = useTranslation('footer')
 
   return (
-    <footer className={styles.container}>
+    <footer data-testid="footer" className={styles.container}>
```
Comment (melinahernandeztheam) [inline src/sections/layout/footer/Footer.tsx:12]:
```suggestion
    <footer className={styles.container}>
```
I think we can remove this data-testid now

File: .eslintrc.json (hunk 1 above)
Comment (melinahernandeztheam) [reply .eslintrc.json:32]: it looks great! thanks! I think there is only 1 missing data-testid that we need to remove

Also, what are we planning to do with the vitest tests inside the /test folder? should we remove them now? or do I remove them in the new coverage issue?

**Review by MellyGray — APPROVED** [review]
Looks great to me 👍 We'll remove the vitest tests in the coverage issue #52  to don't break the coverage now

**Comment by kcondon** [comment]
When I run the component test using npm run cy:run-component I see an error, npm ERR! Missing script: "cy:run-component" . 

What am I doing wrong? I cloned the repo, did git checkout -b feature/add-component-tests
